src/pages/index.js

```diff
@@ -24,24 +24,189 @@ const profileDescriptionInput = document.querySelector(
+      if (userInfo) {
+        document.querySelector(".profile__image").src = userInfo.avatar;
+      } else {
+        console.error("Invalid user information received");
+      }
+    })
+    .catch((error) => {
+      console.error("Error getting user information:", error);
+    });
+});
+
+/* Profile Image Click Handler */
+const profileImage = document.querySelector(".profile__image");
+profileImage.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+/* Handlers */
 function handleImageClick(cardData) {
   imagePopup.open(cardData);
 }
+
 function handleProfileEditSubmit(inputValues) {
+  const updatedData = {
+    name: inputValues.name,
+    about: inputValues.about,
+  };
+
+  // Change button text to "Saving..."
+  profilePopup.setLoadingState(true);
+
+  api
+    .updateUserProfile(updatedData)
+    .then((updatedUserInfo) => {
+      userProfile.setUserInfo(updatedUserInfo);
+      console.log("User profile updated successfully:", updatedUserInfo);
+    })
+    .catch((error) => {
+      console.error("Error updating user profile:", error);
+    })
+    .finally(() => {
+      // Reset button text to its original state
+      profilePopup.setLoadingState(false);
+    });
+}
+function handleAvatarSubmit({ avatar }) {
+  // Change button text to "Saving..."
+  avatarPopup.setLoadingState(true);
+
+  api
+    .updateAvatar({ avatar: avatar })
+    .then((updatedUserInfo) => {
+      console.log("Avatar updated successfully:", updatedUserInfo);
+      document.querySelector(".profile__image").src = updatedUserInfo.avatar;
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error);
+    })
+    .finally(() => {
+      // Reset button text to its original state
+      avatarPopup.setLoadingState(false);
+    });
+}
+
+function handleFormSubmit() {
   userProfile.setUserInfo(inputValues);
 }
+
 function handleAddCardFormSubmit({ name, link }) {
-  const card = renderCard({ name, link }, cardListEl);
-  cardList.addItem(card);
+  // Change button text to "Saving..."
+  cardPopup.setLoadingState(true);
+
+  api
+    .createCard({ name, link })
+    .then((createdCard) => {
+      console.log("Card created successfully", createdCard);
+      const cardElement = renderCard(createdCard);
+      cardList.addItem(cardElement);
+    })
+    .catch((error) => {
+      console.error("Error creating card", error);
+    })
+    .finally(() => {
+      // Reset button text to its original state
+      cardPopup.setLoadingState(false);
+    });
+}
+
+function handleLike(card) {
+  api
+    .likeCard(card.getId())
+    .then((updatedCard) => {
+      console.log("Card liked successfully", updatedCard);
+      card._isLiked = true;
+      card._likeButton.classList.add("card__like-button_active");
+    })
+    .catch((error) => {
+      console.error("Error liking card", error);
+    });
+}
+
+function handleDislike(card) {
+  api
+    .dislikeCard(card.getId())
```

Comment:
**Could be improved**

To have access to any method of the class in `index.js` you can pass `this` to the call:

```js

        this._likeButton.addEventListener('click', () => {
            this._handleLike(this);
        });


```

And now in `index.js` you can call the methods like this:


```js


function addLike(cardInstance) {
        api.likeCard(cardInstance.getId())
            .then((res) => {
               cardInstance.addLike(res)
            })
            .catch((err) => {
                console.error(err);
            });
}


```

The same thing is with deleting cards